### ISSUE TITLE
question: removing folders at major update uninstall phase 

### ISSUE THREAD

**Issue by kot-pilot** [oleg-shilo/wixsharp#705]
when we have a major update msi is doing uninstall ->install and when it uninstalls - product directories and files are deleted. Is there a way to stop it from removing files/folders at update uninstall phase, but if we are just uninstalling a product not while updating it should still  remove them. My problem is associated with situation when i need some config files to be saved while update ( like web config ,loggers, etc)

**Comment by AwesomeM4rc** [comment]
Hi,

I don't know if it is possible to stop it from removing product directories and files but I also needed to save some files while updateing. 

I used the Managed Setup and did something like that to Safe and Write Back my Files -> 

    private static void Msi_BeforeInstall(SetupEventArgs e)
        {
            if(e.IsUninstalling){
                if(e.IsUpgrading){
                    //Save Files to Tempfolder
                }
            }
        }

    private static void Msi_AfterInstall(SetupEventArgs e)
        {
            if(e.IsInstalling){
                if(Directory.Exist(tempFolder))
                //Copy saved Files to Product Directory
                //Delete Temp Directory
            }
        }

Inside the Msi_AfterInstall I checked for the temp Folder to exist and only if it exists I do the Copy to the Product Directory. So in Order:

1. On Uninstall the old MSI (Version 1.0 for example) will safe the files to the Tempfolder

2. On Install the new MSI (Version 2.0 for example) will see that the Tempfolder exists and therefore will copy the files to your Product Directory ( I used File.Copy with overwrite parameter "true").

But maybe there are better soultions for this issue.
